App center documentation (#238)
exoplatform/exo-documentation: docs/Administration.rst

-  The number of added users is increased if some users were added to the bound group by an administrator

.. _AdminAppCenter:

============================
Managing Application Center
============================

As a platform administrator, you can manage the Application Center by:

-  Defining the number of allowed favorite applications
-  Defining mandatory applications that appear in every favorite applications drawer/list
-  Defining default icon for added applications
-  Adding/Editing applications

To access the administration menu for Application Center, you need to access to the left navigation menu through

> the hamburger icon -> Administration -> OTHER -> Applications

Reviewer: The menu label changed. Please update the path

Author: ok

Author: the navigation location has been changed, the label OTHER is still the same, I'll keep it as is until it is synchronized with crowdin

> |AC_image0|

Reviewer: The label of the menu changed recently. Please update the image to reflect the correct label

Author: OK

Settings tab
~~~~~~~~~~~~~

|AC_image10|

In the settings tab you can define the two properties:

-  Number of favorites allowed: which defines the :ref:`maximum number of applications <max-fav-apps>` an end user is able to favorite

-  Default app icon: which defines the icon that could take an application when it is added without an icon

To define the above properties, you just need to click on the corresponding edit icon |AC_image2| in front of 
the property you want to redefine.

For the property **Number of favorites allowed**, you need either to move the cursor or click on +/- icons to reach
the desired value then click on save icon |AC_image4|.

|AC_image3|

To set a custom default icon for application, other the one defined by eXo Platform, you need to follow these steps:

-  click on the edit icon 
-  browse for an image in your machine
-  click on save icon to load the image in its location

|AC_image5|

If you need to edit the image again, you will find the previous image name instead of the browse button with a delete
button allowing to restore the default icon.

|AC_image6|

.. _ApplicationsTabAppCenter:

Applications tab
~~~~~~~~~~~~~~~~~

|AC_image11|

> Applications tab contains:

Reviewer: The preview of this paragraph is KO.
The 2 dash are displayed on the same line, not on different lines

Author: Ok I'll check

-  a table with the whole applications list with delete and edit actions for each line (i.e. each application)
-  ``Add an application`` button

In this table, the three radio button properties: ``Mandatory``, ``Active`` and ``Mobile``,  
are editable through the table. You just need to switch the radio button.

.. _EditApp:

Edit action
------------

When you click on the edit action, a drawer opens containing the already seized properties.

|AC_image12|

.. note:: For system applications, i.e. those installed in the platform, name, URL and image fields
          are not editable. They appear grey colored in the edit mode.
          
          |AC_image13|
          
.. _DeleteApp:

Delete action
---------------

When you click on the delete action, a popup appears to confirm the application's deletion.

|AC_image14|

> Clicking on ``Delete`` button confirms the deletions, clicking on ``Cancel`` cancels teh deletion and nothing happens.

Reviewer: ```suggestion
Clicking on ``Delete`` button confirms the deletion, clicking on ``Cancel`` cancels the deletion and nothing happens.
```

~~~~~~~~~~~~~~~~~~~~

In the :ref:`Applications tab <ApplicationsTabAppCenter>`, you have a button "Add new application" |AC_image7|
allowing you to open the "Add application" drawer with these fields:

|AC_image9|

-  ``Title``: a mandatory field allowing to define the application's name
-  ``URL``: a mandatory field allowing to set the access URL of the application
-  ``Image``: not mandatory, allows you to define an icon for the application, you can browse for it through your files, it
   should have a maximum size of 100 ko, otherwise, it will not be accepted.
-  ``Description``: not mandatory, allows you to set a description for your application to explain more its usage. 
   This field can contain a maximum of 1000 characters.
-  ``Mandatory``: Allows you to define if the :ref:`application is mandatory <mandatoryApps>` for end users or not, 
   by default set to NO   
-  ``Active``: Allows you to define if the application should be enabled or not i.e. displayed to its 
   audience (defined in the ``Permissions`` field), by default set to "YES" i.e. active
-  ``Mobile``: Allows you to enable or disbale the application's display in mobile devices, i.e. if you judge it a not

>    mobile friendly app, you can set this option to false and that way ot will not be dispaled in mobile devices. 

Reviewer: ```suggestion
   mobile friendly app, you can set this option to false and that way ot will not be displayed in mobile devices. 
```

   By default set to true.

> -  ``Permissions``: to define the application's audience, By default set to *:/platform users i.e. if you don't define

Reviewer: *:/platform users generates a link on the * character.
Can we remove it ?

Author: ok

   this field, the added application will be displayed for the platform users
-  ``Help page``: Allows you to set a help page URL, it could be any page containing more details about the application
   or a how to guide. If you define this field, a :ref:`clickable question mark <helpPage>` will be added to the 
   application's card and redirecting to the defined help page.
   

> .. note:: The |AC_image8| button remains on disabled status until mandatory fields (``Title`` and ``URL``) are set.

Reviewer: AC_image8 is missing and broken in the preview

Author: ok

-  The number of added users is increased if some users were added to the bound group by an administrator

.. _AdminAppCenter:

============================
Managing Application Center
============================

As a platform administrator, you can manage the Application Center by:

> -  Defining the number of allowed favorite applications

Reviewer: ```suggestion
-  Defining the number of allowed favorites applications
```
Not sure that favorite take a 's'

> -  Defining mandatory applications that appear in every favorite applications drawer/list

Reviewer: ```suggestion
-  Defining mandatories applications that appear in every favorites applications drawer/list
```
